rts/Lua/LuaSyncedRead.cpp

```diff
@@ -1631,10 +1631,19 @@ int LuaSyncedRead::GetAllyTeamList(lua_State* L)
 
 
 /***
- *
+ * Get all team IDs.
+ * 
+ * @function Spring.GetTeamList
+ * @param allyTeamID -1|nil (Default: `-1`) 
```

Comment:
shouldn't it be
```diff
- -1|nil
+ number|nil
```

Reply:
Nope. If you pass `-1` or `nil` you get an array. Any other number is `integer[]|nil` as the return value. Check the overload above it.

Basically it's smart enough to detect which overload you're using and fix the return type based on it. So if you do `GetTeamList(5)` it will force you to `nil` check because there may be no team 5. But if you do `GetTeamList()` or `GetTeamList(-1)` it will use this overload you've commented on which always returns array.